app/models/cinstance.rb

```diff
@@ -148,7 +148,9 @@ def validate_plan_is_unique?
   }
 
   def self.provided_by(account)
-    joins(:service).references(:service).merge(Service.of_account(account)).readonly(false)
+    # we can access service through plan but also keep service.id in sync with plan.service.id
+    # this is a simpler way to do the query used historically
+    joins(:service).where.has { service.sift(:of_account, account) }
```

Comment:
This is the main fix point (UPDATED). But there are further optimizations in other files.

Reply:
@akostadinov Do you have any idea why the ogiginal query had `readonly(false)`?

Reply:
no idea, probably itdidn't work without it at some point. I think it shouldn't be needed anymore

https://stackoverflow.com/a/33360519/520567